src/components/structures/auth/Registration.tsx

```diff
@@ -139,8 +139,21 @@ export default class Registration extends React.Component<IProps, IState> {
 
     componentDidMount() {
         this.replaceClient(this.props.serverConfig);
+        //triggers a confirmation dialog for data loss before page unloads/refreshes
+        window.addEventListener("beforeunload", this.unloadCallback);
     }
 
+    componentWillUnmount() {
+        window.removeEventListener("beforeunload", this.unloadCallback);
+    }
+
+    private unloadCallback = (event: BeforeUnloadEvent) => {
+        event.preventDefault();
```

Comment:
This will always show the confirmation dialog

> According to the specification, to show the confirmation dialog an event handler should call [preventDefault()](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault) on the event.

(Empirically only in Firefox & Safari)